### PR TITLE
Added blob public access and CORS rule options

### PR DESCRIPTION
**Pull request by adamtodd0**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PUB-1011

### Change description ###
Added options for blob public access and CORS rules. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -144,4 +149,23 @@ variable "tables" {
   type        = list(string)
   description = "List of Storage Tables"
   default     = []
+}
+
+// CORS
+variable "cors" {
```
Comment (chris-pateman) [inline variables.tf:155]: Can you add some validation for the 'allowed_methods' please.
https://www.terraform.io/language/values/variables#custom-validation-rules

File: main.tf [hunk 2]
```diff
@@ -40,6 +42,13 @@ resource "azurerm_storage_account" "storage_account" {
       delete_retention_policy {
         days = 365
       }
+	  cors_rule {
```
Comment (chris-pateman) [inline main.tf:45]: Can this block be dynamic depending if the 'allowed_origins' has any inputs please

**Review by chris-pateman — CHANGES_REQUESTED** [review]

File: main.tf [hunk 3]
```diff
@@ -40,6 +42,16 @@ resource "azurerm_storage_account" "storage_account" {
       delete_retention_policy {
         days = 365
       }
+	  dynamic cors_rule {
+		for_each = var.cors_rules
+		content {
+			allowed_headers = var.cors_rules.allowed_headers
```
Comment (chris-pateman) [inline main.tf:48]: to access the value from the loop for_each you will need to access the properties like `each,value.allowed_headers` 
https://www.terraform.io/language/meta-arguments/for_each

**Review by chris-pateman — CHANGES_REQUESTED** [review]

**Review by chris-pateman — APPROVED** [review]
LGTM

**Review by timja — APPROVED** [review]

**Pull request merged**